### ISSUE TITLE
CTRL-C should trigger cleanup process, including sending SIGINT/kill to all tunnels

### ISSUE THREAD

**Issue by mdsauce** [maxdobeck/sauced#26]
Currently if you CTRL-C the main process it will trigger the local cleanup and kill the clients but not trigger the tunnel's API call to cleanup.

**Comment by mdsauce** [comment]
tunnels on sauce will die after 5 minutes

**Comment by mdsauce** [comment]
FYI a how-to:
https://stackoverflow.com/questions/11268943/is-it-possible-to-capture-a-ctrlc-signal-and-run-a-cleanup-function-in-a-defe

**Comment by mdsauce** [comment]
Will just have to run all stop logic:
```
c := make(chan os.Signal, 1)
signal.Notify(c, os.Interrupt)
go func(){
    for sig := range c {
        // sig is a ^C, handle it
       // send out stop all command to the tunnels here
    }
}()
```

**Comment by mdsauce** [comment]
added logic that stops tunnels locally just fine.  ~But because the parent sauced process doesn't wait properly for waitgroups to return we end up killing the tunnels hard.  May not wait for jobs to complete if we kill this way. Adding a note for user in there now.~

Should be closed once #33 is solved.  Blocked by #33 on the `sigint-trap` branch.
